### ISSUE TITLE
Authentication not working when app is deployed to Openshift

### ISSUE THREAD

**Issue by tarekshumaila** [jedireza/drywall#311]
Hi All,

I am having a weird issue with authentication. I am using Mongolab as my database, authentication request, password reset and new user registration are working fine locally, but when I deployed the app to Openshift, I get a proxy  error whenever I try to login. I get two different errors, one that is 503 (Service Temporarily Unavailable) - when I type the login pair correctly and 502 (proxy error) when I type the login pair incorrectly. 

Also, my content is coming from the Mongolab database and it's showing up fine, so I am assuming connection to the db is ok. Just the authentication piece that is not working. I am using the local strategy and no social media logins are used.

Maybe I am missing a setting? I will attach a picture of the errors and my node app dependencies. Please help.

Thanks!

![screen shot 2015-07-13 at 2 13 09 am](https://cloud.githubusercontent.com/assets/6101167/8645512/33066eca-2905-11e5-94f1-6edfc05b9f55.png)

![screen shot 2015-07-13 at 2 16 05 am](https://cloud.githubusercontent.com/assets/6101167/8645534/6579abce-2905-11e5-94ee-0caa0dc9bbef.png)

**Comment by jedireza** [comment]
There are quite a few results for [openshift 502 proxy error](https://www.google.com/search?q=openshift+502+proxy+error) in Google. Please ddouble check that this isn't an OpenShift environment issue.

**Comment by tarekshumaila** [comment]
Thanks jedireza, I dug deeper and found my issue. I will specify it here just incase someone else runs into the same problem. Basically the problem was around bcrypt module, when I deployed on Mac OSX Yosemite it was working, but because I carried over the modules to Openshift, it wasn't functioning properly with linux. The solution:

1. delete the entire project, spin up a new gear
2. clone the new repo and add your node app to it. 
3. add a .gitingore file and write this to it in the first line "node_modules/*" save.
4. follow openshift instruction to update https://developers.openshift.com/en/managing-modifying-applications.html

doing this will let Openshift install the bcrypt module for linux. More details on the solution found in this article:

http://stackoverflow.com/questions/15809611/bcrypt-invalid-elf-header-when-running-node-app/20590261#20590261

take away lesson: use the .gitignore for node_modules going forward and let package.json do the job to install modules for you.

Cheers!
